platform/mellanox/integration-scripts/sdk_kernel_patches.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import shutil
+import argparse
+import copy
+import difflib
+
+from helper import *
+
+class Data:
+    # old sonic-linux-kernel series file 
+    old_series = list()
+    # updated sonic-linux-kernel series file
+    new_series = list()
+    # list of any new SDK patches to add
+    new_patches = list()
+    # list of current patch list in sonic-linux-kernel
+    old_patches = list()
+    # index of the mlnx_hw_mgmt patches start marker in old_series
+    i_sdk_start = -1 
+    # index of the mlnx_hw_mgmt patches end marker in old_series
+    i_sdk_end = -1
+    # kernel directory to consider the patches
+    k_dir = ""
+
+class SDKAction(Action):
+    def __init__(self, args):
+        super().__init__(args)
+
+    def read_data(self):
+        Data.old_series = FileHandler.read_raw(os.path.join(self.args.build_root, SLK_SERIES))
+    
+    def find_sdk_patches(self):
+        (Data.i_sdk_start, Data.i_sdk_end) = FileHandler.find_marker_indices(Data.old_series, SDK_MARKER)
+        if Data.i_sdk_start < 0 or Data.i_sdk_end > len(Data.old_series):
+            print("-> FATAL mellanox_sdk marker not found. Couldn't continue.. exiting")
+            sys.exit(1)
+
+        print("-> INFO mellanox_sdk markers found. start: {}, end: {}".format(Data.i_sdk_start, Data.i_sdk_end))
+        for index in range(Data.i_sdk_start+1, Data.i_sdk_end):
+            patch = Data.old_series[index].strip()
+            if patch:
+                Data.old_patches.append(Data.old_series[index].strip())
+        print("-> INFO Current mellanox sdk upstream patches. \n{}".format("\n".join(Data.old_patches)))
+
+    def get_kernel_dir(self):
+        # Get the kernel dir name to get the patches
+        try:
+            (kernel, major, minor) = self.args.sonic_kernel_ver.split(".")
+        except Exception as e:
+            print("-> FATAL Kernel formatting error: " + str(e))
+            sys.exit(1)
+        
+        Data.k_dir = os.path.join(KERNEL_BACKPORTS, "{}.{}.{}".format(kernel, major, minor))
+        if os.path.exists(os.path.join(self.args.patches, Data.k_dir)):
+            return 
+        
+        # if the k_dir with actual minor doesn't exit, use the patches generated against kernel.major
+        Data.k_dir = os.path.join(KERNEL_BACKPORTS, "{}.{}.{}".format(kernel, major, "0"))
+        path_to_check = os.path.join(self.args.patches, Data.k_dir)
+        if os.path.exists(path_to_check):
+            return 
+        else:
+            print("-> FATAL Kernel dir with patches doesn't exist: {}".format(path_to_check))
+            sys.exit(1)
+
+    def filter_patches(self):
```

Comment:
Should we rename this to process_patches?

Reply:
Updated